exonum-java-binding-core/src/test/java/com/exonum/binding/storage/indices/ProofMapIndexProxyIntegrationTest.java

```diff
@@ -545,7 +547,37 @@ public void getMultiProof_SortedMultiEntryMapContains() {
       List<MapEntry<HashCode, String>> entries = createSortedMapEntries();
       putAll(map, entries);
 
-      assertThat(map, provesThatCorrect(entries));
+      assertThat(map, provesThatPresent(entries));
+    });
+  }
+
+  @Test
+  public void getMultiProof_FourEntryMap_LastByte_DoesNotContain1() {
```

Comment:
We test an empty map here — not a 4 entry map, and it is not that useful as the proof structure is always the same for an empty map. What is interesting is testing absent keys in a map that contains something, varying the common prefix between requested keys and nodes of the tree:
```
            <00xxxx>
            /        \
    <00|00xx>          <00|10xx>
   /         \        /         \
<0000|01>  <0000|11>  <0010|00>   <0010|10>
```

and keys:
- 010000 — rejected at root level
- 0001xx, 0011xx - rejected at intermediate level
- 000000, 000010, 001001, 001011 — rejected at leaf level

Please double-check the tree above.

Reply:
Will do first thing tomorrow!

Reply:
I'd also include the illustration.

Please remember that it shows **bit** strings (the same as in DbKey*ParameterizedTest).